src/Driver/PostGISPlatform.php

```diff
@@ -116,28 +107,24 @@ public function getCreateTablesSQL(array $tables): array
     public function getAlterTableSQL(TableDiff $diff): array
     {
         $table = $diff->getOldTable();
-        $spatialIndexes = [];
         $spatialIndexSqlGenerator = new SpatialIndexSqlGenerator($this);
 
-        if ($table) {
-            $spatialIndexes = SpatialIndexes::ensureTableDiffFlag($diff);
-        }
-
-        SpatialIndexes::filterTableDiff($diff);
+        SpatialIndexes::ensureSpatialIndexFlags($diff);
 
-        $sql = parent::getAlterTableSQL($diff);
+        $sql = parent::getAlterTableSQL(SpatialIndexes::filterTableDiff($diff));
 
-        if (!$table) {
-            return $sql;
+        foreach (SpatialIndexes::extractSpatialIndicies($diff->getAddedIndexes()) as $spatialIndex) {
+            $sql[] = $spatialIndexSqlGenerator->getSql($spatialIndex, $table);
         }
 
-        foreach ($spatialIndexes as $spatialIndex) {
-            $sql[] = $spatialIndexSqlGenerator->getSql($spatialIndex, $table);
+        foreach (SpatialIndexes::extractSpatialIndicies($diff->getModifiedIndexes()) as $index) {
+            $sql[] = $this->getDropIndexSQL($index->getName(), $table->getName());
+            $sql[] = $spatialIndexSqlGenerator->getSql($index, $table);
         }
 
         /** @var ColumnDiff $columnDiff */
         foreach ($diff->getModifiedColumns() as $columnDiff) {
-            if ($columnDiff->hasChanged('srid')) {
+            if ($columnDiff->getOldColumn()->getPlatformOption('srid') !== $columnDiff->getNewColumn()->getPlatformOption('srid')) {
```

Comment:
This throws an exception when the 'srid' option is not defined (example: the column is not one from PosGis)

I would add something like this before the check:

```php
            if( ! $columnDiff->getNewColumn()->hasPlatformOption('srid')) {
                continue;
            }
```

Reply:
The error is thrown when running `bin/console doctrine:schema:validate -v`

Reply:
Thanks so much for the testing! Fix pushed.

Reply:
Thank you for the PR! :+1: 

Unfortunately DBAL v4 is not (yet) compatible with Symfony, thus my tests will be limited to v3 only from now on